File: MB-Chat/providers/provider-fallback.ts
import { buildProviderResponse } from "./provider-response";
import type { ProviderRequest, ProviderResponse } from "./types";

export function buildSafeProviderFallback(request: ProviderRequest, reason: string, retryCount = 0): ProviderResponse {
  return buildProviderResponse({
    request,
    provider_name: "none",
    model_name: "safe-fallback",
    status: "fallback",
    content: "",
    latency_ms: 0,
    safety_flags: ["SAFE_FALLBACK", reason],
    fallback_used: true,
    retry_count: retryCount,
  });
}

File: MB-Whatsapp/providers_py/provider_fallback.py
from __future__ import annotations

from .provider_response import build_provider_response
from .types import ProviderRequest, ProviderResponse


def build_safe_provider_fallback(request: ProviderRequest, reason: str, retry_count: int = 0) -> ProviderResponse:
    return build_provider_response(
        request,
        provider_name="none",
        model_name="safe-fallback",
        status="fallback",
        safety_flags=["SAFE_FALLBACK", reason],
        fallback_used=True,
        retry_count=retry_count,
    )
